### ISSUE TITLE
CTE with IN

### ISSUE THREAD

**Issue by polderudo** [sqlc-dev/sqlc#2153]
### Version

1.17.2

### What happened?

Using IN with recursive CTE gives "relation does not exist".

On the added playground you can see that all queries, except GetAll3 do work. 
The query is working in Postgres however.
Also GetAll4 works, if using a known table name as CTE name.

### Database schema

```sql
CREATE TABLE L (
  id   BIGSERIAL PRIMARY KEY,
  parent_id int null
);

CREATE TABLE T (
  id   BIGSERIAL PRIMARY KEY,
  L_ID int,
  F varchar(256)
);
```

### SQL queries

```sql
-- name: GetAll :many
SELECT * FROM L;

-- name: GetAll1 :many
with recursive cte as (
  select id, L_ID, F from T
  union all
  select c.id, c.L_ID, c.F from T as c where c.L_ID = $1
) select id, l_id, f from cte;

-- name: GetAll2 :many
with recursive cte as (
  select id, L_ID, F from T where T.ID=2
  union all
  select c.id, c.L_ID, c.F from T as c where c.L_ID = $1
) select id, l_id, f from cte;

-- name: GetAll4 :many
select id from T where L_ID in(
  with recursive L as (
    select id, L_ID, F from T where T.ID =2
    union all
    select c.id, c.L_ID, c.F from T as c where c.L_ID = $1
 ) select l_id from L
);

-- name: GetAll3 :many
select id from T where L_ID in(
  with recursive cte as (
    select id, L_ID, F from T where T.ID =2
    union all
    select c.id, c.L_ID, c.F from T as c where c.L_ID = $1
 ) select l_id from cte
);
```

### Configuration

```json
{
  "version": "1",
  "packages": [
    {
      "path": "db",
      "engine": "postgresql",
      "schema": "query.sql",
      "queries": "query.sql"
    }
  ]
}
```

### Playground URL

https://play.sqlc.dev/p/22128fd8cad468350179fc3f1efb2b6b019505e507331ecb7362f924401753bb

### What operating system are you using?

macOS

### What database engines are you using?

PostgreSQL

### What type of code are you generating?

Go

**Comment by kyleconroy** [comment]
I had to update the query in a few places to get it working with PostgreSQL, but can confirm that our current analyzer chokes on `GetAll3`.

**Comment by kyleconroy** [comment]
This is fixed in [v1.23.0](https://docs.sqlc.dev/en/latest/reference/changelog.html#v1-23-0) by enabling the [database-backed query analyzer](https://docs.sqlc.dev/en/latest/howto/generate.html#enhanced-analysis-with-managed-databases). We added a [test case](https://github.com/sqlc-dev/sqlc/tree/main/internal/endtoend/testdata/cte_with_in/postgresql/pganalyze) for this issue so it won’t break in the future.

You can play around with the [working example on the playground](https://play.sqlc.dev/p/cc83dc4241dab014354fae745afed471629de393d82fbbe9256397ff615f82e7)
